pyquil/gates.py

```diff
@@ -36,14 +47,15 @@ def unpack_reg_val_pair(classical_reg1, classical_reg2):
     """
     left = unpack_classical_reg(classical_reg1)
     if isinstance(classical_reg2, int) or isinstance(classical_reg2, float):
-        right = classical_reg2
-    else:
-        right = unpack_classical_reg(classical_reg2)
-
-    return left, right
+        return left, classical_reg2
+    return left, unpack_classical_reg(classical_reg2)
 
 
-def prepare_ternary_operands(classical_reg1, classical_reg2, classical_reg3):
+def prepare_ternary_operands(classical_reg1: MemoryReferenceDesignator,
+                             classical_reg2: MemoryReferenceDesignator,
+                             # TODO:(appleby) why doesn't prepare_ternary_operands allow floats for classical_reg3
```

Comment:
This appears to be a bug? According to https://github.com/rigetti/quil/blob/master/spec/typed-memory.md, all of the classical ops that use this function to unpack their args (`EQ`, `GT`, `GE`, ...) can take an immediate real as the third argument.